eodatasets3/prepare/sentinel_l1c_prepare.py

```diff
@@ -278,32 +319,61 @@ def prepare_and_write(
     "--output-base",
     help="Write metadata files into a directory instead of alongside each dataset",
     required=False,
-    type=PathPath(exists=True, writable=True, dir_okay=True, file_okay=False),
+    type=PathPath(
+        exists=True, writable=True, dir_okay=True, file_okay=False, resolve_path=True
+    ),
 )
 def main(
     output_base: Optional[Path],
     datasets: List[Path],
     overwrite_existing: bool,
 ):
-    for dataset in datasets:
 
-        if dataset.is_dir():
-            output_path = dataset / f"{dataset.stem}.odc-metadata.yaml"
-        else:
-            output_path = dataset.with_suffix(".odc-metadata.yaml")
+    logging.basicConfig(
+        format="%(asctime)s %(levelname)s %(message)s", level=logging.INFO
+    )
+    for input_path in datasets:
+        in_out_paths = [
+            *(
+                (
+                    "sinergise.com",
+                    # Input is the folder
+                    p.parent,
+                    # Output is an inner metadata file, with the same name as the folder (usually S2A....).
+                    (p.parent / f"{p.parent.stem}.odc-metadata.yaml"),
+                )
+                for p in _rglob_with_self(input_path, "tileInfo.json")
+            ),
+            *(
+                (
+                    "esa.int",
+                    # Input is zip file
+                    p,
+                    # Output is a sibling file with metadata suffix.
+                    p.with_suffix(".odc-metadata.yaml"),
+                )
+                for p in _rglob_with_self(input_path, "*.zip")
+            ),
+        ]
```

Comment:
Can this part be turned into a function and have some unit tests?  The tests could also be useful to show examples of where the meta-data.yaml should be located to work.

Reply:
It could! my gut feeling is that the bang-for-buck is too low, as the important logic is exercised by the integration tests already ("call it with input filename, expect output filename", etc), and unit test code here would be larger than the function but not give us much extra useful coverage. 

(I'm sometimes hesitant with unit tests as they tend to be higher maintence than integration tests -- they depend too much on internal implementation structure, so are broken easily. There's quite a few in eodatasets, but they're usually used when low cost to write, high reward in what they cover, or test something extremely unlikely to change)